### DISCUSSION TITLE
From VIPS colorspace to RGB888 ?

### DISCUSSION THREAD

**Discussion by ademmler** [libvips/libvips#2030] [Q&A]
How do I convert from VIPs internal colorspace to RGB888?
Or how can I set "bits/sample" (and other options) while writing out files.
Can you point me to the right chapter in documentation.

The resulting tiff image from vips says:
  Bits/Sample: 32
  Sample Format: IEEE floating point
  Compression Scheme: None
  Photometric Interpretation: RGB color
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3

This is what I would need:
  Bits/Sample: 8
  Sample Format: unsigned integer
  Compression Scheme: None
  Photometric Interpretation: RGB color
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3

**Comment by jcupitt** [comment]
Try:

```python
image = image.cast("uchar")
```

To cast down to uint8.

I suppose this is a TIFF file? libvips automatically casts down to the format supported by the output file on save, but of course TIFF supports every number format you can think of, so you'll get a float TIFF if you have a float image.

**Comment by ademmler** [comment]
ok - what I am really struggling is, that Qt expects a certain image format to create a pixmap.
Lets say it is RGB888. After doing vips processing the image formate seems to be changed.
Thats what I get from a file I have saved for debugging.

This is (snippet) what I am doing. The displayed image is wired again.

vipsObject = VImage::new_from_file (imagePath.toStdString().c_str());   
       vipsObject = vipsObject.pow(1.00);       
       vipsObject.cast(VIPS_FORMAT_UCHAR);

       imageObject = new QImage();       
       *imageObject = QImage((uchar*)vipsObject.data(), vipsObject.width(), vipsObject.height(),    VIPS_IMAGE_SIZEOF_LINE(vipsObject.get_image()), QImage::Format_RGB888);
       image = QPixmap::fromImage(*imageObject);

**Comment by jcupitt** [comment]
Try:

```C++
vipsObject = vipsObject.cast(VIPS_FORMAT_UCHAR);
```

Or perhaps:

```C++
vipsObject = vipsObject.pow(1.0).cast(VIPS_FORMAT_UCHAR);
```

libvips images are all immutable (they can't be changed), so all libvips methods are functions that return a new image.

**Comment by ademmler** [comment]
ok - that was very helpful - thx - is there any book about libvips?

**Comment by jcupitt** [comment]
Just the docs:

https://libvips.github.io/libvips/API/current/

The "overview" at the start runs through most of the main features.

**Comment by ademmler** [comment]
Perfekt ... slowley I get how it works ... thx so much.
